Add a try catch in Log Task#run() to catch errors that occur and let LogCron continue planning for future logging.

reda-idtaleb opened the pull request.

reda-idtaleb committed:
add a try catch in LogTask#run() method to catch errors occured during
running a log task

reda-idtaleb committed:
include the debug log into the try catch in LogTask#run()

pmouawad commented on back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java:
```diff
@@ -52,9 +52,13 @@ public LogTask(
-		this.registry.logAndReset();
-		System.out.println(this.registry.guiLog());
+		try {
+			LOG.debug("Sample buffer : {}",this.sampleQueue.size());
```
Add LOG.isDebugEnabled()

pmouawad commented on back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java:
```diff
@@ -52,9 +52,13 @@ public LogTask(
+		try {
+			LOG.debug("Sample buffer : {}",this.sampleQueue.size());
+			this.registry.logAndReset();
+			System.out.println(this.registry.guiLog());
```
Can you replace this with LOG.info

reda-idtaleb replied:
ok

pmouawad commented on back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/LogTask.java:
Can you rename the class so that it reflects its real responsability ?

reda-idtaleb replied:
This is done in an other branch (#96_RI) because the renaming was related to the #96 issue.

pmouawad commented on back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java:
```diff
@@ -217,7 +217,7 @@ public SampleLog makeLog(String name, Date timestamp) {
 		
 		return currentPeriodSummary == null ? null : new SampleLog(
-				Util.micrometerToOpenMetrics(name),
+				Util.makeOpenMetricsName(name),
```
Is this related to the PR ? or should you rebase  your branch ?

reda-idtaleb replied:
Actually, this branch was already merged to the develop branch. And I commited in the same merged branch, then i did the PR . I shouldn't do that ? Should I create a new branch ?

pmouawad commented on back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/listener/ULPObservabilityListener.java:
```diff
@@ -224,7 +224,7 @@ public void setRegex(String regex) {
 
 	public String getTotalLabel() {
-		return getPropertyAsString(ULPODefaultConfig.TOTAL_LABEL_PROP, ULPODefaultConfig.totalLabel());
+		return Util.makeOpenMetricsName(getPropertyAsString(ULPODefaultConfig.TOTAL_LABEL_PROP, ULPODefaultConfig.totalLabel()));
```
Is this related to the PR ? or should you rebase  your branch ?

reda-idtaleb replied:
The rebase is done. I took some times to understand what it do because i never used it. Now the PR should only has the new commits.

reda-idtaleb committed:
change log.debug to Log.info in LogTask#run()

reda-idtaleb committed:
Refactroring : rename the class LogTask to SampleMetricsLoggingTask and

reda-idtaleb committed:
Revert "Refactroring : rename the class LogTask to SampleMetricsLoggingTask and"

This reverts commit b2c6bcab1f69e9949370304c7b7efbd00a75ca53.

reda-idtaleb committed:
Refactoring: rename the class LogTask to SampleMetricsLoggingTask and
change it in all the callers

reda-idtaleb committed:
remove the verification of the logFrequency on the
ULPObservabilityListener#init() because the logFrequecy is always
strictly positive

reda-idtaleb committed:
Change the default value of the logFrequecny to 30 seconds

reda-idtaleb committed:
LogFrequency defaults to 30 second when its value is set to zero on the GUI (#107)

* Refactroring : rename the class LogTask to SampleMetricsLoggingTask and

* Revert "Refactroring : rename the class LogTask to SampleMetricsLoggingTask and"

This reverts commit b2c6bcab1f69e9949370304c7b7efbd00a75ca53.

* Refactoring: rename the class LogTask to SampleMetricsLoggingTask and
change it in all the callers

* remove the verification of the logFrequency on the
ULPObservabilityListener#init() because the logFrequecy is always
strictly positive

* Change the default value of the logFrequecny to 30 seconds

This resolves #96

pmouawad commented on back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java:
```diff
@@ -52,9 +52,13 @@ public LogTask(
+			this.registry.logAndReset();
+			System.out.println(this.registry.guiLog());
+		} catch (Exception e) {
+			LOG.error("Error while running the log task : {}", e);
```
Isn't there a missing parameter for {}

pmouawad commented on back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java:
```diff
@@ -52,9 +52,13 @@ public LogTask(
+		try {
+			LOG.info("Sample buffer : {}",this.sampleQueue.size());
+			this.registry.logAndReset();
+			System.out.println(this.registry.guiLog());
```
Can you log as info instead of using System.out ?

pmouawad commented on back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/task/SampleMetricsLoggingTask.java:
```diff
@@ -52,9 +52,13 @@ public LogTask(
-		System.out.println(this.registry.guiLog());
+		try {
+			LOG.info("Sample buffer : {}",this.sampleQueue.size());
+			this.registry.logAndReset();
```
Can you rename logAndReset to computeAndReset ?

pmouawad reviewed (changes requested):
Thanks for PR, see my comments.

reda-idtaleb committed:
change the name of MicrometerResgistry#logAndReset()
to #computeAndReset()

reda-idtaleb committed:
MicrometerResgitered#logAndReset() changed to computeAndReset() and
MicrometerResgitered#log() changed to logSampleMetrics()

reda-idtaleb committed:
Merge branch '96_RI' into 83_RI

reda-idtaleb commented:
> Thanks for PR, see my comments.

Done
